src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
```

Comment:
[FUNCTION NAMING] - The `hide` method should only set the `hidden` property to `true`. Currently, it toggles the value, which could lead to unintended behavior where calling `hide` could actually unhide the herbivore. Consider changing the implementation to simply set `this.hidden = true;`.